sn_cli/src/operations/config.rs

```diff
@@ -494,12 +495,31 @@ impl Config {
         Ok(())
     }
 
-    pub async fn write_prefix_map(path: &Path, prefix_map: &NetworkPrefixMap) -> Result<()> {
+    pub async fn write_prefix_map(path: &PathBuf, prefix_map: &NetworkPrefixMap) -> Result<()> {
+        let prefix_map_dir = path
+            .parent()
+            .ok_or_else(|| eyre!("Path {:?} should be inside a folder", path.display()))?;
+        let temp_file = NamedTempFile::new_in(prefix_map_dir)
+            .wrap_err_with(|| "Error creating temp file".to_string())?;
+
         let serialized = Self::serialise_prefix_map(prefix_map)?;
-        fs::write(path, serialized).await.wrap_err_with(|| {
-            format!("Unable to write NetworkPrefixMap to '{}'", path.display())
+        fs::write(temp_file.path(), serialized)
+            .await
```

Comment:
We have the file opened when `temp_file` was instantiated, so we can just write to it directly with `write_all`. `fs::write` will re-open the file and then write.

```suggestion
        temp_file.write_all(serialized)
```

Reply:
I have made the changes, but my thought was to make the write async, `tempfile` is sync only. So I assume the cost of opening the file > sync writes?

Reply:
>So I assume the cost of opening the file > sync writes?

Possibly, but it's so infrequent that I don't think it will even register.